x-pack/plugins/enterprise_search/server/routes/app_search/credentials.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+import { IRouteDependencies } from '../../plugin';
+import { createEnterpriseSearchRequestHandler } from '../../lib/enterprise_search_request_handler';
+
+interface ICredentialsResponse {
+  results: object[];
+  meta?: {
+    page?: {
+      total_results: number;
+    };
+  };
+}
```

Comment:
Oooh. I really like this type def being at the top of the page here to give devs an example of what JSON body we expect

Reply:
Hadn't considered that, great point.